From f627a442350560e50dbbb11f7828b6d807369533 Mon Sep 17 00:00:00 2001
From: Alexei Colin <acolin@isi.edu>
Date: Sun, 11 Apr 2021 20:40:00 -0400
Subject: [PATCH] cmake: PETSc,SLEPc: match lowercase .pc pkg-config files

Upstream has moved to lowercase. PETSc has kept the mixed-case file for
compatibility, but SLEPc hasn't.

pkg_search_module takes multiple patterns and succeeds on first match,
so this commit is backward-compatible with older installations of PETSc,
SLEPc.

PETSC: https://gitlab.com/petsc/petsc/-/merge_requests/2754

	commit 6fe8eb356ce6cf9027a2b889b1835f808039fe5e
	Merge: fd0ee0fbd7 2eefe1c6f1
	Author: Satish Balay <balay@mcs.anl.gov>
	Date:   Tue Jun 9 01:08:49 2020 +0000

	    Merge branch 'MR--config-lowercase-pkgname' into 'master'

	    config: pkg-config: default to lowercase file.pc

	    See merge request petsc/petsc!2754

SLEPc: https://gitlab.com/slepc/slepc/-/merge_requests/68

	commit 1632350b0a681e0c4fd08c171912be6e17f3c3f7
	Merge: 3c38a3419 b2fcc7199
	Author: Jose E. Roman <jroman@dsic.upv.es>
	Date:   Sun Jun 14 16:19:22 2020 +0000

	    Merge branch 'jose/pkg-config' into 'master'

	    pkg-config: default to lowercase file.pc

	    See merge request slepc/slepc!68

Signed-off-by: Alexei Colin <acolin@isi.edu>
---
 cmake/modules/FindPETSc.cmake | 2 +-
 cmake/modules/FindSLEPc.cmake | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/modules/FindPETSc.cmake b/cmake/modules/FindPETSc.cmake
index 80faffdad..728a6ec30 100644
--- a/cmake/modules/FindPETSc.cmake
+++ b/cmake/modules/FindPETSc.cmake
@@ -51,7 +51,7 @@ find_package(PkgConfig REQUIRED)
 
 # Find PETSc pkg-config file. Note: craypetsc_real is on Cray systems
 set(ENV{PKG_CONFIG_PATH} "$ENV{CRAY_PETSC_PREFIX_DIR}/lib/pkgconfig:$ENV{PETSC_DIR}/$ENV{PETSC_ARCH}/lib/pkgconfig:$ENV{PETSC_DIR}/lib/pkgconfig:$ENV{PKG_CONFIG_PATH}")
-pkg_search_module(PETSC craypetsc_real PETSc)
+pkg_search_module(PETSC craypetsc_real petsc PETSc)
 
 # Extract major, minor, etc from version string
 if (PETSC_VERSION)
diff --git a/cmake/modules/FindSLEPc.cmake b/cmake/modules/FindSLEPc.cmake
index 327cffc23..dec26fcc4 100644
--- a/cmake/modules/FindSLEPc.cmake
+++ b/cmake/modules/FindSLEPc.cmake
@@ -48,7 +48,7 @@ find_package(PkgConfig REQUIRED)
 set(ENV{PKG_CONFIG_PATH} "$ENV{SLEPC_DIR}/$ENV{PETSC_ARCH}/lib/pkgconfig:$ENV{SLEPC_DIR}/lib/pkgconfig:$ENV{PKG_CONFIG_PATH}")
 set(ENV{PKG_CONFIG_PATH} "$ENV{PETSC_DIR}/$ENV{PETSC_ARCH}/lib/pkgconfig:$ENV{PETSC_DIR}/lib/pkgconfig:$ENV{PKG_CONFIG_PATH}")
 set(ENV{PKG_CONFIG_PATH} "$ENV{PETSC_DIR}/$ENV{PETSC_ARCH}:$ENV{PETSC_DIR}:$ENV{PKG_CONFIG_PATH}")
-pkg_search_module(SLEPC crayslepc_real SLEPc)
+pkg_search_module(SLEPC crayslepc_real slepc SLEPc)
 
 # Extract major, minor, etc from version string
 if (SLEPC_VERSION)
-- 
2.30.0

